### PR TITLE
fix(jest-warnings): updates jest and other dependencies

### PR DESCRIPTION
**Pull request by priley86**
**What**:
This PR updates Jest and fixes some existing warnings with React 16 / request animation frame polyfill. This polyfill now ships with Enzyme, so I've gone ahead included it. We can setup other Enzyme packages/tests later...

I've also removed `babel-preset-es2015` and `babel-preset-stage-0` in favor of `babel-preset-env` and babel-plugin-tranforms.

The Jest `modulePathIgnorePatterns` can also be used to fix warnings with jest-haste. https://github.com/facebook/jest/issues/2070

Deployed storybook (just in case):
https://rawgit.com/priley86/patternfly-react/jest-warnings-fixed/index.html

### CODE REVIEW

**Comment by priley86** [comment]
cc: @ohadlevy @mturley

File: package.json [hunk 1]
```diff
@@ -76,11 +81,20 @@
...
     "storybook": "start-storybook -c storybook -p 6006",
     "build-storybook": "build-storybook -c storybook -o .out",
     "storybook:deploy": "storybook-to-ghpages",
     "semantic-release": "semantic-release pre && npm publish && semantic-release post"
   },
+  "jest": {
+    "modulePathIgnorePatterns": [
+      "lib"
+    ],
+    "setupFiles": [
```
Comment (ohadlevy) [inline package.json]: Does this works with older versions of react?

Comment (priley86) [reply package.json]: I misread the [docs](http://facebook.github.io/jest/docs/en/configuration.html#modulepathignorepatterns-array-string) originally. It should be `["<rootDir>/lib/"] instead.

It seems React `^15.6.2` and Jest `^19.0.2` can still accept `modulePathIgnorePatterns`. I did a simple test in this [branch](https://github.com/priley86/patternfly-react/tree/react-15-test) in my fork. I think it is only dev dependency though, so not sure it matters. I am only adding this to remove the warnings about duplicate manual mock when Jest runs currently in this repo...

**Comment by priley86** [comment]
I've updated this PR and the storybook w/ all of the dependency updates noted in #89 along w/ updating React to 16.2

@jtomasek I've also manually added the `lib` to a branch in my fork. Can you test w/ this branch in Triple O and let me know what issues you find? You will also need to update the ListView I think...

```
"patternfly-react": "https://github.com/priley86/patternfly-react.git#deps-test",
```

**Comment by priley86** [comment]
@jtomasek out sailing/cruising the seas? ⛵️  🛥

**Comment by priley86** [comment]
will rebase this later today...

**Comment by priley86** [comment]
repurposed this in #152

**Pull request closed without merging**